pkg/test/test.go

```diff
@@ -0,0 +1,178 @@
+package test
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	conformanceImage         = "k8s.gcr.io/conformance:corev1.18.6"
+	conformanceNamespaceName = "conformance"
+	conformanceResourceName  = "conformance"
+)
+
+type Config struct {
+	Config *rest.Config
+}
+
+type Test struct {
+	k8sClient kubernetes.Interface
+}
+
+func New(config Config) (*Test, error) {
+	k8sClient, err := kubernetes.NewForConfig(config.Config)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	test := Test{
+		k8sClient: k8sClient,
+	}
+
+	return &test, nil
+}
+
+func (t *Test) RunAWSConformance() error {
+	return nil
+}
+
+func (t *Test) RunKubernetesConformance(ctx context.Context) error {
```

Comment:
Copied from https://raw.githubusercontent.com/kubernetes/kubernetes/master/cluster/images/conformance/conformance-e2e.yaml I'm not sure if this will actually work--it's a placeholder.

Reply:
Just to make the intent here clear for reviewers, all test logic here will ultimately go away entirely and the runner "test" will simply start a Pod which does the actual test